llvm/include/llvm/Support/RISCVISAInfo.h

```diff
@@ -22,7 +23,8 @@ struct RISCVExtensionInfo {
   unsigned MinorVersion;
 };
 
-void riscvExtensionsHelp();
+void getAllExtensions(StringMap<RISCVExtensionInfo> &Features);
```

Comment:
This can go if you just set the description of only one extension in the tests.

Reply:
I would leave it.

It's fine for endusers to have an API with an accessor to the *full* list of what RISCV have.
Whole mine effort here started with lack of accessor to exactly such thing: https://github.com/llvm/llvm-project/pull/66582

Reply:
Regarding accesors see [these lines]
(https://github.com/cbalint13/tvm/blob/4c17af1a187dee450ab89107df18623e8edadede/src/target/llvm/llvm_module.cc#L90-L107) in an external project.

--

The info desired there would be usefull in super simple queries like: "Is true that user desired CPU X have feature X", e.g. a concrete example, a user have a ```casadelake``` and check up if it is ```avx512bw``` capable, and also LLVM backend have it implemented or not (if it is older).

Reply:
Ok but I'll refer you to MaskRay's comment there:
```
We need justification to have this API, ideally tested by an in-tree target, otherwise this is prone to be removed by another contributor as an unused function.
```

There is no need to add a dummy description to every extension just for the test. Just pick one that already exists. Then there's no in tree use for this function and MaskRay's logic applies again.

The fact that you were able to add such a method without much trouble also tells us that carrying it in tree, unused, isn't going to save anyone much effort when it finally is needed.

Reply:
An external project needing something is not a reason to add methods in tree. It could be part of an argument to do so but in the context of this patch, this new RISC-V method is not needed and should be removed.

If you want to ease TVM's pain by upstreaming their changes you can do so, in a PR focused on that, with some in tree use case and testing.

Reply:
It was men't as example. Anyway, prior certain versions (in our case 18) that accessor is still needed, won't go anywhere soon for compatibility.

Reply:
I remove it (all), and add only partial fillers.